pkg/sidecar/sidecar.go

```diff
@@ -153,6 +155,22 @@ func (s *Sidecar) updateCertificates(svidResponse *workloadapi.X509Context) {
 // signalProcess sends the configured Renew signal to the process running the proxy
 // to reload itself so that the proxy uses the new SVID
 func (s *Sidecar) signalProcess() (err error) {
+	if s.config.PidFileName != "" {
+		atomic.StoreInt32(&s.processRunning, 1)
```

Comment:
why 1 is required inside processRunning?
it looks like it is to avoid if that is set bellow, 
if that is the case I think we msut avoid this in order to simply the code and add this if as part of the switch, and run a signal there...

Reply:
Yeah. Really wasn't sure how to redo that switch, when honestly I wasn't exactly sure how it was working to begin with. Setting it to 1 let it fall through to the right path.

How should this be reworked?

Reply:
I took a stab at reworking it.